lib/jxl/enc_frame.cc

```diff
@@ -1167,9 +1163,27 @@ Status EncodeGlobalACInfo(PassesEncoderState* enc_state, BitWriter* writer,
     if (enc_state->cparams.decoding_speed_tier >= 1) {
       hist_params.max_histograms = 6;
     }
+    size_t num_histogram_groups = shared.num_histograms;
+    if (enc_state->streaming_mode) {
+      size_t prev_num_histograms =
+          enc_state->passes[i].codes.encoding_info.size();
+      if (enc_state->update_global_state) {
+        prev_num_histograms += kNumFixedHistograms;
```

Comment:
Unless I misunderstand the code, it seems that the `update_global_state` flag here is doing "double duty" to also detect the first iteration. Is that incorrect?

Reply:
Yes, maybe this is not the best name, this flag guards the things that we have to do when we first fill in structures that are then used at subsequent DC groups.

Reply:
initialize_global_state perhaps? Makes the "do it once" part clearer...

Reply:
Done.